File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
    const map = new Map();
    const res=[];
    for (let i = 0; i < strs.length; i++) {
        let rotatedstr = strs[i].split("").sort().join("");
        
        if (!map.has(rotatedstr)) {
            map.set(rotatedstr, [strs[i]])
        }
        else {
            let arr = map.get(rotatedstr)
            arr.push(strs[i])
            map.set(rotatedstr, arr)
        }
    }
    console.log(map)
    map.forEach((value,key)=>{
       
        res.push(value)
    })
    return res
};

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mymap;
        vector<vector<string>> result;
        int n = strs.size();
        for(int i=0;i<n;i++){
            string sortedvalue = strs[i];
            sort(sortedvalue.begin(), sortedvalue.end());
            mymap[sortedvalue].push_back(strs[i]);
        }
        for(auto i:mymap){
            result.push_back(i.second);
        }
        return result;
    }
};